src/main.js

```diff
@@ -162,12 +167,15 @@ module.exports = {
           await waitOnIdle()
         }
         const textEditor = atom.workspace.getActiveTextEditor()
+        const text = textEditor.getText()
+        if (text.length === 0) {
+          return
+        }
         const filePath = textEditor.getPath()
         const projectPath = atom.project.relativizePath(filePath)[0]
 
-        if (!textEditor || textEditor.isModified()) {
+        if (!textEditor) {
           // Abort for invalid or unsaved text editors
-          atom.notifications.addError('Linter-ESLint: Please save before fixing')
```

Comment:
There is a small caveat here.  If you make a change to introduce a linting error within a previously-passing file without saving, and the fix results in the same code as what was previously saved, manually running a fix command will have no visible effect in the editor.  That's because the file on disk will essentially not be changed, and so Atom won't know to reload the editor.  I think that's a small enough edge case that we don't need to worry about it, but wanted to make you aware.